rust/libnewsboat/src/utils.rs

```diff
@@ -2,6 +2,30 @@ pub fn replace_all(input: String, from: &str, to: &str) -> String {
     input.replace(from, to)
 }
 
+pub fn consolidate_whitespace( input: String ) -> String {
+    let found = input.find( |c: char| !c.is_whitespace() );
+    let mut result = String::new();
+
+    if found.is_some() {
+        let (leading,rest) = input.split_at(found.unwrap());
```

Comment:
Please replace that `if` with `if let` so you don't have to call `unwrap`:

```rust
if let Some(found) = found {
    let (leading, rest) = input.split_at(found);
    ...
}
```

Reply:
Thats a neat trick. Thank you.